include/openthread/platform/settings.h

```diff
@@ -68,12 +68,12 @@ enum
     OT_SETTINGS_KEY_RESERVED             = 0x0006, ///< Reserved (previously auto-start).
     OT_SETTINGS_KEY_SLAAC_IID_SECRET_KEY = 0x0007, ///< SLAAC key to generate semantically opaque IID.
     OT_SETTINGS_KEY_DAD_INFO             = 0x0008, ///< Duplicate Address Detection (DAD) information.
-    OT_SETTINGS_KEY_LEGACY_OMR_PREFIX    = 0x0009, ///< Reserved. Legacy Off-mesh routable (OMR) prefix.
-    OT_SETTINGS_KEY_ON_LINK_PREFIX       = 0x000a, ///< On-link prefix for infrastructure link.
+    OT_SETTINGS_KEY_RESERVED2            = 0x0009, ///< Reserved (previously off-mesh routable (OMR) prefix).
+    OT_SETTINGS_KEY_RESERVED3            = 0x000a, ///< Reserved (previously on-link prefix for infrastructure link).
```

Comment:
I would prefer keeping the name and adding `LEGACY`, `DEPRECATED` or `RESERVED` into the name.

Reply:
`DEPRECATED` might be a good choice.

Reply:
SGTM, thanks!